hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -953,9 +955,25 @@ private void finishActiveMasterInitialization(MonitoredTask status)
     if (!waitForMetaOnline()) {
       return;
     }
+    TableDescriptor metaDescriptor = tableDescriptors.get(
+        TableName.META_TABLE_NAME);
+    final ColumnFamilyDescriptor tableFamilyDesc = metaDescriptor
+        .getColumnFamily(HConstants.TABLE_FAMILY);
+    final ColumnFamilyDescriptor replBarrierFamilyDesc =
+        metaDescriptor.getColumnFamily(HConstants.REPLICATION_BARRIER_FAMILY);
+
     this.assignmentManager.joinCluster();
     // The below depends on hbase:meta being online.
-    this.tableStateManager.start();
+    try {
+      this.tableStateManager.start();
+    } catch (NoSuchColumnFamilyException e) {
+      if (tableFamilyDesc == null && replBarrierFamilyDesc == null) {
+        LOG.info("For missing CFs in meta, this Exception is expected", e);
```

Comment:
You think operator will know what 'missing CFs in meta' is about? Would it be better to talk about migration from hbase1 to hbase2?